litestar/testing/client/base.py

```diff
@@ -83,10 +99,16 @@ def __init__(
                 UserWarning,
                 stacklevel=1,
             )
+
         self._session_backend: BaseSessionBackend | None = None
         if session_config:
             self._session_backend = session_config._backend_class(config=session_config)
-        self.app = app
+
+        if not isinstance(app, Litestar):
+            app = _add_state_app(app)  # type: ignore[assignment]
+
+        self.app = cast("T", app)  # type: ignore[redundant-cast]  # pyright needs this
```

Comment:
We depend on the `"state"` key existing in `scope`, and setting it is one of the first things that we do with every connection:

https://github.com/litestar-org/litestar/blob/a2e5b7854edce5b40d23bc3493d47f579bdcddef/litestar/app.py#L541

However, `httpx` doesn't inject `state` with its scope ([ref](https://github.com/encode/httpx/blob/87f39f12c9ecdd0ebcd8d017235698e72567e9c7/httpx/_transports/asgi.py#L98-L111)):

```py
    async def handle_async_request(
        self,
        request: Request,
    ) -> Response:
        assert isinstance(request.stream, AsyncByteStream)

        # ASGI scope.
        scope = {
            "type": "http",
            "asgi": {"version": "3.0"},
            "http_version": "1.1",
            "method": request.method,
            "headers": [(k.lower(), v) for (k, v) in request.headers.raw],
            "scheme": request.url.scheme,
            "path": request.url.path,
            "raw_path": request.url.raw_path,
            "query_string": request.url.query,
            "server": (request.url.host, request.url.port),
            "client": self.client,
            "root_path": self.root_path,
        }
```
... that scope is constructed totally within that function scope and there is no interface to add extra keys to it.

This is why I'm wrapping arbitrary non-litestar asgi-apps with another app that injects the "state" key into the scope.

Reply:
There's only one pattern of tests in our own test suite that this affects, that is the one where we construct a simple asgi app and pass it to `TestClient` and then use the scope to construct a connection instance, like this:

https://github.com/litestar-org/litestar/blob/8e4b50ceb79de292410d8d9600ed15000191ecb7/tests/unit/test_connection/test_request.py#L158-L167

So, while I'm not over the moon about this, the alternative is to do something like `scope.setdefault("state", {})` in `ASGIConnection.__init__()`, and I don't like having to change to library code just to support running tests.